src/storage/MonitoringStore.ts

```diff
@@ -29,31 +30,38 @@ export class MonitoringStore<T extends ResourceStore = ResourceStore>
   }
 
   public async addResource(container: ResourceIdentifier, representation: Representation,
-    conditions?: Conditions): Promise<ResourceIdentifier> {
+    conditions?: Conditions): Promise<ModifiedResource> {
     const identifier = await this.source.addResource(container, representation, conditions);
-    this.emitChanged([ container, identifier ]);
+    this.emitChanged([ changedResource(container), identifier ]);
```

Comment:
I'd suggest a different design, namely:
- Emit specific `created`, `deleted`, `changed` events, with the identifier of the resource as a parameter
- Still also emit the `changed` event, with the identifier as a first parameter, and the kind of change as the second

That way:
- The `change` mechanism keeps on working as-is, and listeners can keep on being interested in any change
- Listeners can choose to only follow certain events

Reply:
> Emit specific created, deleted, updated events, with the identifier of the resource as a parameter

We initially tried implementing the event emission in that way, but we realised that valuable information would be lost when when an operation affects multiple resources in different ways(e.g. some `created` and some `updated`). Due to this we thought wrapping all these changes into an object would be better. Therefore we implemented the `ModifiedResource` to wrap all performed resource changes. *I now recognize based on @joachimvh review comment that this could have been better achieved by using the already existing class `RepresentationMetadata`.*

> Still also emit the changed event, with the identifier as a first parameter, and the kind of change as the second

Good point, for backwards compatability. 👍